source/exe/terminate_handler.cc

```diff
@@ -2,14 +2,28 @@
 
 #include <cstdlib>
 
+#include "envoy/common/exception.h"
 #include "source/common/common/logger.h"
 #include "source/server/backtrace.h"
 
+#include "absl/strings/str_format.h"
+
 namespace Envoy {
 
 std::terminate_handler TerminateHandler::logOnTerminate() const {
   return std::set_terminate([]() {
-    ENVOY_LOG(critical, "std::terminate called! (possible uncaught exception, see trace)");
+    auto currentException = std::current_exception();
+    if (currentException) {
+      try {
+        std::rethrow_exception(currentException);
+      } catch (const EnvoyException& e) {
+        ENVOY_LOG(critical, absl::StrFormat("std::terminate called! Uncaught EnvoyException '%s', see trace.", e.what()));
+      } catch (...) {
+        ENVOY_LOG(critical, "std::terminate called! See trace. Uncaught unknown exception, see trace.");
```

Comment:
can't you capture the std::exception as a fallback here and also call e.what() for more detail?

Reply:
oh...maybe you have to catch std::exception in a block first so you have something to call "what()" on. In what scenarios would the "..." clause be reached if you catch std::exception first?

Reply:
> In what scenarios would the "..." clause be reached if you catch std::exception first?

When the thrown exception isn't derived from std::exception, which is unlikely but possible.

Reply:
Added logging of exceptions derived from std::exception